### PR TITLE
--context support to calicoctl

### PR DESCRIPTION
**Pull request by frozenprocess**
Using https://github.com/projectcalico/libcalico-go/pull/1335 this PR adds `--context` argument to calicoctl.
## Todos
- [ ] Tests
- [x] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by frozenprocess** [comment]
I was not sure if there is a way to pass arguments into libcalico [Client](https://github.com/projectcalico/calicoctl/blob/be82d923e77efb5c44f8489ba55ada67f3bf2e83/calicoctl/commands/clientmgr/client.go#L31) so I decided to convert `--context` value into an environment variable.

@caseydavenport @robbrockbank

**Comment by robbrockbank** [comment]
Hi Reza,

I would have included the field value in the CalicoAPIConfigSpec and then fill in the configOverrides.

see https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/k8s/k8s.go#L220

Also, you might need to change how we handle using the "default" namespace. We have this hardcoded in calicoctl here: https://github.com/projectcalico/calicoctl/blob/master/calicoctl/commands/common/resources.go#L361

In terms of where the new option lives.  I would put it at the same level as `--config`.

@caseydavenport :  are you ok with these suggestions?

File: calicoctl/commands/delete.go [hunk 1]
```diff
@@ -16,7 +16,8 @@ package commands
 
 import (
 	"fmt"
-	"strings"
+  "strings"
```
Comment (caseydavenport) [inline calicoctl/commands/delete.go:19]: Looks like you'll need to run `gofmt` on this file

File: calicoctl/commands/get.go [hunk 2]
```diff
@@ -61,7 +62,8 @@ Options:
   -a --all-namespaces          If present, list the requested object(s) across all namespaces.
   --export                     If present, returns the requested object(s) stripped of
                                cluster-specific information. This flag will be ignored
-			       if <NAME> is not specified.
```
Comment (caseydavenport) [inline calicoctl/commands/get.go:64]: Indentation looks wrong here too? Although seems it was wrong in the other direction to begin with :D

**Comment by caseydavenport** [comment]
> @caseydavenport : are you ok with these suggestions?

Yep, that all sounds right to me

File: calicoctl/commands/get.go (hunk 2 above)
Comment (frozenprocess) [reply calicoctl/commands/get.go:64]: Nice catch!

**Comment by frozenprocess** [comment]
> I would have included the field value in the CalicoAPIConfigSpec and then fill in the configOverrides.
About this one

I've added a new string value `K8sCurrentContext` to `apiconfig.go` file in `kubeconfig` section for this specific override, since context is only going to affect `kubeconfig` scenarios I thought it would be best to do it in that section.

I talked to @robbrockbank about this one and he seemed to be satisfied with my explanation.


>In terms of where the new option lives. I would put it at the same level as --config.
My last commit supports `--context` regardless of its position. (more like kubectl)
```
calicoctl --context mycontext
calicoctl get --context mycontext node
calicoctl get node --context mycontext
```
@caseydavenport WDYT?

File: calicoctl/commands/delete.go (hunk 1 above)
Comment (frozenprocess) [reply calicoctl/commands/delete.go]: I Love `gofmt`
Thank you!

**Comment by frozenprocess** [comment]
Added test cases to check `--context` in multiple scenarios.

Extended `CalicoctlMayFail` function in `tests/fv/utils/calicoctl.go` to support multicluster. (`KUBECONFIG` got a lower priority than`K8S_API_ENDPOINT` that is why this change was needed.)

**Comment by frozenprocess** [comment]
I've updated my base commit to #2223.

**Review by caseydavenport — APPROVED** [review]
@frozenprocess this looks good to me! I think you need to run `gofmt` though, there are one or two places the spacing looks a bit off.

**Pull request merged**